### PR TITLE
Feat watch later

### LINKED ISSUE #15: Todo: Watch Later Feature

### PR DESCRIPTION
**Pull request by coldpigli**
1) Added Watch Later Page
2) The functionality to add or remove videos from watchlist

image

### CODE REVIEW

File: src/components/VideoInfo/VideoInfo.jsx [hunk 1]
```diff
@@ -1,81 +1,36 @@
 import styles from "./VideoInfo.module.css";
 import { useState } from "react";
 import { useUserDetails } from "../../contexts";
-import { checkLogin, toast } from "../../utils";
-import axios from "axios";
+import { checkLogin } from "../../utils";
+import { handleLikeVideo, handleUnlikeVideo } from "../../utils/handleLikeUnlike";
+import { addToWatchlater, removeFromWatchlater } from "../../utils/handleWatchLater";
```
Comment (rahulcs754) [inline src/components/VideoInfo/VideoInfo.jsx:6]: you can make a index file and call in one line

File: src/components/VideoItem/VideoItem.jsx [hunk 2]
```diff
@@ -1,67 +1,23 @@
-import axios from "axios";
-import React from "react";
 import { Link } from "react-router-dom";
 import { useUserDetails } from "../../contexts";
-import { checkLogin, toast } from "../../utils";
+import { handleLikeVideo, handleUnlikeVideo } from "../../utils/handleLikeUnlike";
+import { addToWatchlater, removeFromWatchlater } from "../../utils/handleWatchLater";
```
Comment (rahulcs754) [inline src/components/VideoItem/VideoItem.jsx:4]: same above

File: src/utils/handleLikeUnlike.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          "/api/user/likes",
+          { video: videoData },
+          {
+            headers: {
+              authorization: localStorage.getItem(tokenName),
+            },
+          }
+        );
+        if (res.status === 200 || res.status === 201) {
+          const { likes } = res.data;
```
Comment (rahulcs754) [inline src/utils/handleLikeUnlike.js:18]: you have to no need this line you can do on line number 8 like this.

const { data : {likes}, status }

**Review by rahulcs754 — APPROVED** [review]
The code is good and clean.
just small suggestion

**Review by krishanu2911 — APPROVED** [review]
No changes required well written code
well defined function understandable code
Good to go👍🏻

**Pull request merged**